Deployment Guides/Provisioning-Islandora-on-Ubuntu-16.md

```diff
@@ -0,0 +1,1796 @@
+
+You should run `update-alternatives --config php` and make sure it is set to `php7.1`. Else Drupal will not install. This is because we are using the 7.1 Apache module and 7.2 material may well have been installed and set to the default for the OS during the `apt-get install`.
+
+Run `mysql_secure_installation`. You will be prompted for the root password and answers to various questions. If you have created a strong root password at this time, feel free to enable the password validation plugin and set the strength requirements appropriately. You can always do this again later if you wish to use a weak password during the setup phase. Answer yes to all the other questions.
+
+### Software Dependencies Compiled from Source <a id="from-source"></a> 
+
+#### ghostscript  <a id="ghostscript"></a>
+
+**Notes:** Ubuntu 14.04 installs ghostscript 9.10 which is currently failing our test sets and prevents tiffs from being generated from pdfs. Recommend installing from source to get correct version. Ubuntu 16.04 installs ghostscript 9.18, which appears to work. So the following is optional.
+```
+cd ~
+
+wget http://downloads.ghostscript.com/public/old-gs-releases/ghostscript-9.05.tar.gz
+
+tar xvzf ghostscript-9.05.tar.gz
+
+cd ghostscript-9.05
+  
+./configure
+ 
+make && make install
+
+`gs --version` #should return result.
+
+```
+
+ffmpeg build location
+```
+mkdir ~/ffmpeg-source
+cd ~/ffmpeg-source
+```
+#### libvpx <a id="libvpx"></a>
+```
+git clone https://chromium.googlesource.com/webm/libvpx.git
+
+cd libvpx
+
+git checkout 8366a6e4ba95e6d5af040815d2afbb4bfe628d3f
+
+./configure --disable-examples --disable-unit-tests
+
+make
+
+checkinstall --pkgname=libvpx --pkgversion="1:$(date +%Y%m%d%H%M)-git" --backup=no --deldoc=yes --fstrans=no --default
+```
+
+#### ffmpeg <a id="ffmpeg"></a>
+```
+cd ~/ffmpeg-source
+
+wget http://www.ffmpeg.org/releases/ffmpeg-1.1.1.tar.gz
+
+tar xf ffmpeg-1.1.1.tar.gz && rm -rf ffmpeg-1.1.1.tar.gz  
+
+cd ffmpeg-1.1.1
+```
+
+Had to do this to remove strange characters that cause build errors.
+```
+sed -i 's/×/x/' doc/filters.texi
+
+sed -i 's/×/x/' doc/ffmpeg.texi
+
+./configure --enable-gpl --enable-libass --enable-libfaac --enable-libfdk-aac --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-librtmp --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx264 --enable-nonfree --enable-version3 --enable-libopus
+
+make
+
+checkinstall --pkgname=ffmpeg --pkgversion="7:$(date +%Y%m%d%H%M)-git" --backup=no --deldoc=yes --fstrans=no --default
+
+make install
+
+hash -r
+```
+
+The following should tell apt to not upgrade ffmpeg (we need version 1.1.1):  
```

Comment:
We have actually tested up to 3.3.x and it works fine just have not had the opportunity to update deployment guide

Reply:
Updated to 3.3.7.

Reply:
Thank you!

Reply:
Forgot to point out. The newer ffmpeg doesn't support libfaac. https://trac.ffmpeg.org/wiki/Encode/AAC

MP4s won't work right if you don't change the codec:

drush -y vset islandora_video_mp4_audio_codec "libfdk_aac"

Reply:
Fixed a typo in the ffmpeg section and added this drush command
in the later Drupal configuration section.